src/spikeinterface/sortingcomponents/peak_pipeline.py

```diff
@@ -211,11 +275,30 @@ def compute(self, traces, peaks):
         return sparse_wfs
 
 
+def find_parent_of_type(node, parent_type, unique=True):
```

Comment:
I agree with what @alejoe91 said about this yesterday.

The code could simply return a list of the parents with the desired type.

`[node for node in parents if instance(node, parent_type)]`

And is the responsability of the each node class to test for uniquness. That follows the single responsability principle. That said, if you agree with this I can take care of this refactor later together with the changes in the waveform extractor. That is, I don't think this is a critical part of this PR.

Reply:
Check the code in https://github.com/SpikeInterface/spikeinterface/blob/4cea780f4576ffb920e8afee358ff5ca468bf5a2/src/spikeinterface/sortingcomponents/waveforms/neural_network_denoiser.py#L22-L26

Reply:
I known that it can be one line.
But the rules of finding the correct will be more and more complex (multi parent, one parent sevarl out), so I think it is really important to have a central function somwhere instead of copy/paste a one line logic (`[node for node in parents if instance(node, parent_type)]`) for instance unique or parents.

If you want topropose a refactoring for that function list comprehension based I am OK with it.
I made explicit loop to open the door to the complexcity.

Reply:
I don't feel very strong about this. Your reasoning makes sense. Maybe let's wait for more cases to see what abstraction make sense.